### PR TITLE
Upgrade graphql compiler and client to fix underlying pagination issues

### PR DESCRIPTION
**Pull request by minasmart**
This upgrades the graphql-js-client-compiler to hoist query variables, instead of dynamically defining them at every usage.

It also upgrades the graphql-js-client version to handle generated pagination queries with variables and named fragments (which were previously being ignored in generated pagination queries).

This is hopefully the last thing that needs to be addressed before `v1.0` gets launched.

### CODE REVIEW

File: src/graphql/CollectionsProductsFragment.graphql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 fragment CollectionsProductsFragment on Collection {
-  products(first: $productsFirst) {
+  products(first: 20) {
```
Comment (swalkinshaw) [inline src/graphql/CollectionsProductsFragment.graphql]: we want to hardcode this value?

Comment (minasmart) [reply src/graphql/CollectionsProductsFragment.graphql]: fragments have no access to variables.

Comment (minasmart) [reply src/graphql/CollectionsProductsFragment.graphql]: This wasn't initially valid, it just passed the parser somehow.

I'm not sure how many layers deep we want to parameterize things. Ideally, if people want heavy customization, people should be using the storefront API directly? But this is totally just opinion.

Comment (harisaurus) [reply src/graphql/CollectionsProductsFragment.graphql]: is there a default that shopify uses across other places? if not, 20 seems reasonable to me

Comment (minasmart) [reply src/graphql/CollectionsProductsFragment.graphql]: There is on some of our APIs. The point of the change here was to get stuff to compile.

I can open these params up later by taking this out of a fragment.

**Pull request merged**